### ISSUE TITLE
Settings Page shift L/R around

### ISSUE THREAD

**Issue by RubberDucky007** [linuxmint/cinnamon-spices-desklets#944]
### Desklet version/Build date

1.15 (2023-10-30 12:07:39)

### Cinnamon version

5.8.4

### Distribution

Mint 21.2

### Graphics hardware *and* driver used

NVIDIA Corporation GF108 [GeForce GT 620]

### Desklet name and maintainer

diskspace@schorschii @schorschii

### What happened?

The Settings Window has on top left two Bracket-Buttons. Clicking shifts the page horizontally to... the same page again, left button shift out to the left, right button shifts to the rightside. Same content. It works but... Bug or (useless) Feature?

### Other information

![Bildschirmfoto vom 2023-11-04 22-14-59](https://github.com/linuxmint/cinnamon-spices-desklets/assets/48609807/595c3074-a6d2-4ed5-9096-9bc9b1a06c09)

**Comment by rcalixte** [comment]
> The Settings Window has on top left two Bracket-Buttons. Clicking shifts the page horizontally to... the same page again, left button shift out to the left, right button shifts to the rightside. Same content. It works but... Bug or (useless) Feature?

This means you have multiple instances of the same desklet. The arrows iterate through the settings for each instance. Where you only have a single instance of a desklet, you won't see any arrows. Where you have more than one instance, you are able to configure them all with only one window.

**Comment by schorschii** [comment]
> This means you have multiple instances of the same desklet. The arrows iterate through the settings for each instance. Where you only have a single instance of a desklet, you won't see any arrows. Where you have more than one instance, you are able to configure them all with only one window.

Exactly. This is not a specific feature of my desklet but a general one of the Cinnamon desklet manager. @RubberDucky007 you may have a second instance of the diskspace desklet somewhere on your desktop where you can't see it, e.g. on a second screen which is currently not connected.

You can click on the "Remove All" button in the desklet manager and then add one diskspace desklet again. Now, the arrow buttons should be disappeared since you now only have one desklet instance active.
![Unbenannt](https://github.com/linuxmint/cinnamon-spices-desklets/assets/17480795/8c4a4b76-8447-4fc0-8bba-c54016317beb)

**Comment by RubberDucky007** [comment]
This might be correct. After removing "all" i can restore them and now one shows filesystem and one memory.
BUT: Before i did this there was only ONE instance visible. And i did not deactivated the decorations so it should be visible - on one of the two screens i have active now (just for a few weeks). But there is no second instance i can find. Either via sysguard, nor 'top' 'pstree' or 'ps -axu' i found nothing with the searchwords 'disk' 'space' 'diskspace' or 'desklet'. So, what is the processname or process it run's under? 

I suggest to implement a "show this instance on mainscreen - centered" Menuitem to solve this problem.
The multiple settings-pages are not helpful in this case i think.

**Comment by RubberDucky007** [comment]
> > The Settings Window has on top left two Bracket-Buttons. Clicking shifts the page horizontally to... the same page again, left button shift out to the left, right button shifts to the rightside. Same content. It works but... Bug or (useless) Feature?
> 
> This means you have multiple instances of the same desklet. The arrows iterate through the settings for each instance. Where you only have a single instance of a desklet, you won't see any arrows. Where you have more than one instance, you are able to configure them all with only one window.

That might be possible. But i am unable to find out where the other desklets are, not shown, invisible and no processname to find - and maybe - to kill them separately.
It was not planned and not wanted to start more than one items of this. And then, not realized that there are more than one running. Leeds to this bugreport. -> Suggest to optimize the behavior for it.

**Comment by rcalixte** [comment]
> That might be possible. But i am unable to find out where the other desklets are, not shown, invisible and no processname to find - and maybe - to kill them separately. It was not planned and not wanted to start more than one items of this. And then, not realized that there are more than one running. Leeds to this bugreport. -> Suggest to optimize the behavior for it.

Under the Desklets menu option (or in the System Settings), there is an option to remove a specific desklet. You can use that to remove all of the other instances and then make sure to only add the one you want.
